Tests/ML/utils/test_io_util.py

```diff
@@ -72,27 +72,34 @@ def test_nii_load_zyx(test_output_dirs: OutputFolderForTests) -> None:
 @pytest.mark.parametrize("ground_truth_channel",
                          [None, known_nii_path, f"{good_h5_path}|segmentation|0|1", good_npy_path])
 @pytest.mark.parametrize("mask_channel", [None, known_nii_path, good_npy_path])
+@pytest.mark.parametrize("check_exclusive", [True, False])
 def test_load_images_from_dataset_source(
         metadata: Optional[str],
         image_channel: Optional[str],
         ground_truth_channel: Optional[str],
-        mask_channel: Optional[str]) -> None:
+        mask_channel: Optional[str],
+        check_exclusive: bool) -> None:
     """
     Test if images are loaded as expected from channels
     """
     # metadata, image and GT channels must be present. Mask is optional
     if None in [metadata, image_channel, ground_truth_channel]:
         with pytest.raises(Exception):
-            _test_load_images_from_channels(metadata, image_channel, ground_truth_channel, mask_channel)
+            _test_load_images_from_channels(metadata, image_channel, ground_truth_channel, mask_channel, check_exclusive)
     else:
-        _test_load_images_from_channels(metadata, image_channel, ground_truth_channel, mask_channel)
+        if check_exclusive:
+            with pytest.raises(ValueError):
```

Comment:
I find it super important to always check that it fails for the right reason, for example by checking the error message

Reply:
Agreed. I would normally do this but I was getting an error in running the test so I dropped it. I assumed that was why there is no check in the existing code (e.g. test_io_util line 87-88. I'll try again

Reply:
When I add this:
`
    with pytest.raises(ValueError) as mutually_exclusive_labels_error:
        _test_load_images_from_channels(metadata, image_channel, ground_truth_channel, mask_channel, check_exclusive)
        error_message = str(mutually_exclusive_labels_error.value)
        assert 'not mutually exclusive' in error_message
`
I get this error (locally):

> AssertionError: .value can only be used after the context manager exits